lib/include/ert/ecl/ecl_sum_file_data.hpp

```diff
@@ -0,0 +1,67 @@
+
+#include <ert/util/vector.hpp>
+#include <ert/util/double_vector.h>
+
+#include <ert/ecl/ecl_smspec.hpp>
+#include <ert/ecl/ecl_sum_tstep.hpp>
+#include <ert/ecl/ecl_file.hpp>
+
+namespace ecl {
+
+#define INVALID_MINISTEP_NR -1
+#define INVALID_TIME_T 0
+
+class ecl_sum_file_data_type {
+
+  double  days_start;
+  double  sim_length;
+  int first_report_step;
+  int last_report_step;
+  ecl_smspec_type * smspec;
+  vector_type * data;
+  int_vector_type * report_first_index;
+  int_vector_type * report_last_index;
```

Comment:
Add a params_mapping as a private member here; I think that can be implemented as std::vector<int> instead of int *